Docs/source/Software/drivers.rst

```diff
@@ -11,20 +11,31 @@ Usage
 
 Additional Considerations
 
-Ethernet
+Ethernet 
 ========
 
 Purpose
 
-The Ethernet drivers are sending GPS, CAN, and IMU data to the data acquisition team
+The Ethernet drivers are sending GPS, CAN, and IMU data to the data acquisition team.
 The sunlight is the client while data acquisition is the server
 
 Usage
 
-
+``Ethernet_Init()`` is first called as a way to create the Queue and create a socket when a connection has been established.
+``Ethernet_ConnectToServer()`` is a static blocking function called inside ``Ethernet_Init()`` that is either called when connection
+first needs to be established or when connection is lost with data acquisition.  ``Ethernet_SendMessage()`` then collects data from the 
+Ethernet Queue and sends through Ethernet - if there is no connection, it calls ``Ethernet_ConnectToServer()``. ``Ethernet_SendMessage()`` returns pdFalse
+if the Queue is empty and pdTrue if the Queue is not empty. The struct EthernetMSG_t will hold the data packets of either information 
+regarding IMU, GPS, or CAN. The struct consists of an id enum, length, and a union of data. Since the data is in a union, there will be zero padding
+at the MSB.
```

Comment:
You don't actually talk about how to use the driver in this documentation. There isn't any mention of Ethernet_PutInQueue which is what actually allows the layers above to put messages in the queue. Most of these just discuss what the functions are, not how to use them.